gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala

```diff
@@ -37,6 +37,86 @@ class GcsApiSpec extends FlatSpec with Matchers with OptionValues with EitherVal
   private def stringify(bytes: Stream[IO, Byte]): IO[String] =
     bytes.through(fs2.text.utf8Decode).compile.toChunk.map(_.toArray[String].mkString(""))
 
+  private def assertInitResumableUploadRequest(
+    request: Request[IO],
+    contentLength: Long,
+    expectedSize: Long,
+    chunk: Chunk[Byte],
+    expectedMd5: Option[String]
+  ): Unit = {
+    request.method shouldBe Method.POST
+    request.uri shouldBe initResumableUploadURI
+    request.headers.toList should contain theSameElementsAs List(
+      `Content-Length`.unsafeFromLong(contentLength),
+      `Content-Type`(MediaType.application.json, Charset.`UTF-8`),
+      Header(UploadContentLengthHeader, expectedSize.toString()),
+      Header(UploadContentTypeHeader, "text/event-stream")
+    )
+
+    io.circe.parser
+      .parse(new String(chunk.toArray[Byte]))
+      .right
+      .value shouldBe buildUploadMetadata(path, expectedMd5)
+    ()
+  }
+
+  private def assertInitResumableUploadResponse(
```

Comment:
I don't see much value in separating the assertions from the response generation (if we get an upload initialization request, we always want to run those assertions and then return a matching response). I'd merge this method with the one above.

Same for the other request/response pairs.